app/src/main/java/com/infomaniak/drive/ui/menu/MenuGalleryFragment.kt

```diff
@@ -43,63 +42,66 @@ class MenuGalleryFragment : Fragment() {
     private var binding: FragmentMenuGalleryBinding by safeBinding()
     private val mainViewModel: MainViewModel by activityViewModels()
 
-    private var galleryFragment = GalleryFragment()
-
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
-        binding = FragmentMenuGalleryBinding.inflate(inflater, container, false).apply {
-            swipeRefreshLayout.setOnRefreshListener { galleryFragment.onRefreshGallery() }
-        }
-
-        binding.multiSelectLayout.apply {
-            selectAllButton.isGone = true
-            setMultiSelectClickListeners()
-        }
-
-        return binding.root
-    }
-
-    private fun MultiSelectLayoutBinding.setMultiSelectClickListeners() = with(galleryFragment) {
-        toolbarMultiSelect.setNavigationOnClickListener { closeMultiSelect() }
-        moveButtonMultiSelect.setOnClickListener { onMoveButtonClicked() }
-        deleteButtonMultiSelect.setOnClickListener { deleteFiles() }
-        menuButtonMultiSelect.setOnClickListener {
-            onMenuButtonClicked(
-                multiSelectBottomSheet = GalleryMultiSelectActionsBottomSheetDialog(),
-                areAllFromTheSameFolder = false,
-            )
-        }
+        return FragmentMenuGalleryBinding.inflate(inflater, container, false).also { binding = it }.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
         ViewCompat.requestApplyInsets(binding.galleryListCoordinator)
+        val galleryFragment = addGalleryFragment()
```

Comment:
It's a bit weird to have a `addGaleryFragment` method which is doing 2 things, replacing the fragment AND returning the fragment itself.

Reply:
Something like this?

```kotlin
override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        ViewCompat.requestApplyInsets(binding.galleryListCoordinator)
        initFragment()
    }

    private fun initFragment() {
        getGalleryFragment()?.let { galleryFragment ->
            setUi(galleryFragment)
        } ?: run {
            childFragmentManager.beginTransaction()
                .replace(R.id.galleryFragmentView, GalleryFragment(), GalleryFragment.TAG)
                .commit()
        }
    }

    private fun getGalleryFragment() = (childFragmentManager.findFragmentByTag(GalleryFragment.TAG) as? GalleryFragment)
```

Reply:
Maybe something like this?

```kotlin
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        ViewCompat.requestApplyInsets(binding.galleryListCoordinator)
        initFragment()
    }

    private fun initFragment() {
        val galleryFragment = getGalleryFragment() ?: GalleryFragment().also { displayDefaultFragment(it) }
        setUi(galleryFragment)
    }

    private fun displayDefaultFragment(defaultGalleryFragment: GalleryFragment) {
        childFragmentManager.beginTransaction()
            .replace(R.id.galleryFragmentView, defaultGalleryFragment, GalleryFragment.TAG)
            .commit()
    }
```